Build Error

##### System information (version)
- OpenCV =>master branch
- Operating System / Platform => Linux 9bfa02303e1e 4.4.0-109-generic #132-Ubuntu SMP Tue Jan 9 20:00:40 UTC 2018 ppc64le ppc64le ppc64le GNU/Linux
- Compiler => gcc 5.4 ? (I'm using the Makefile)

##### Detailed description

**When I build cmake, I get the following error after 100%  (It's new, before 7 days it worked correct):**

/usr/bin/ld: CMakeFiles/opencv_python3.dir/__/src2/cv2.cpp.o: In function `pyopencv_cv_cuda_cuda_DeviceInfo_computeMode(_object*, _object*, _object*)':
cv2.cpp:(.text._ZL44pyopencv_cv_cuda_cuda_DeviceInfo_computeModeP7_objectS0_S0_+0x10c): call to `_object* pyopencv_from<unsigned int>(unsigned int const&)' lacks nop, can't restore toc; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
modules/python3/CMakeFiles/opencv_python3.dir/build.make:111: recipe for target 'lib/python3/cv2.cpython-35m-powerpc64le-linux-gnu.so' failed
make[2]: *** [lib/python3/cv2.cpython-35m-powerpc64le-linux-gnu.so] Error 1
CMakeFiles/Makefile2:7500: recipe for target 'modules/python3/CMakeFiles/opencv_python3.dir/all' failed
make[1]: *** [modules/python3/CMakeFiles/opencv_python3.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2


##### Steps to reproduce
git clone https://github.com/opencv/opencv.git
cd opencv/
mkdir build
cd build/
cmake -D CMAKE_BUILD_TYPE=RELEASE ..
make -j5

--- a/modules/python/src2/cv2.cpp
+++ b/modules/python/src2/cv2.cpp
@@ -674,7 +674,7 @@ bool pyopencv_to(PyObject* obj, int& value, const char* name)
     return value != -1 || !PyErr_Occurred();
 }
 
-#if defined (_M_AMD64) || defined (__x86_64__)
+#if defined (_M_AMD64) || defined (__x86_64__) || defined (__PPC64__)
 template<>
 PyObject* pyopencv_from(const unsigned int& value)
 {
@@ -682,7 +682,6 @@ PyObject* pyopencv_from(const unsigned int& value)
 }
 
 template<>
-
 bool pyopencv_to(PyObject* obj, unsigned int& value, const char* name)
 {
     (void)name;

